docs/architecture/ADR-003-borrow-assets.md

```diff
@@ -0,0 +1,121 @@
+# ADR 003: Borrowing assets using uToken collateral
+
+## Changelog
+
+- September 27, 2021: Initial Draft (@toteki)
+
+## Status
+
+Proposed
+
+## Context
+
+One of the base functions of the Umee universal capital facility is to allow users to borrow allowed asset types, using their own uTokens (obtained normally, by depositing assets) as collateral.
+
+## Alternative Approaches
+
+It might be worth considering adding message types which do two or more of the steps below at once (e.g. deposit base asset, convert to uToken collateral, and receive loan all at once). The individual steps to need to be possible in isolation, though.
+
+## Decision
+
+The Cosmos `x/bank` module and the existing `umee/x/leverage` deposit features are prerequisites for these new capabilities.
+
+The flow of events is as follows:
+- Borrower has already used `umee/x/leverage` module to deposit base assets and receive uTokens
+- Borrower deposits uTokens into asset facility as collateral - facility remembers collateral position
+- Borrower requests to borrow assets from capital facility - facility checks request, disburses tokens if acceptable, and remembers borrow position
+- Eventually, borrower pays repayment amount (in full or in part) using borrowed asset denomination
+- Borrower may withdraw collateral uTokens at any time as long as minimum overcollateralization requirement remains satisfied
+- On asset repayment amount = 0 for a given borrow position, facility forgets loan.
+- On uToken collateral amount = 0 for a given collateral position, facility forgets collateral.
```

Comment:
I don't understand these two points. Perhaps consider rephrasing.

Reply:
I'll rephrase it. After the incoming edits, the second line won't exist, and the first will indicate that the `leverage` keeper, which stores all open borrow positions, no longer stores them after they are repaid in full.